### ISSUE TITLE
toHaveLabel() doesn't work with tabBarAccessibilityLabel

### ISSUE THREAD

**Issue by j320** [wix/Detox#2023]
### Description
When adding a tabBarAccessibilityLabel to a bottom navigation tab bar element, locating it by its tabBarTestID and using the toHaveLabel() expectation, the assertion fails (on Android it passes).

#### Reproduction
1. In your app with a bottom navigation tab bar, add a tabBarTestID to one of the sections
2. In the same section, add a tabBarAccessibilityLabel
3. Run a test with the following: 

`expect(element(by.id(tabBarTestID))).toHaveLabel(tabBarAccessibilityLabel)
`
#### Expected behavior
The test passes

#### Environment:
 - Detox: 16.3.0
 - React Native: 0.62.2
 - Node: 14.0.0
 - Device: iPhone 11
 - Xcode: 11.4.1
 - iOS: 13.4.1

**Comment by LeoNatan** [comment]
What is “tabBarAccessibilityID”? What does the view hierarchy show the label as? The only thing that matters for Detox is the native hierarchy, not the JS definitions. If RN doesn’t forward some prop correctly, or does so differently between the platforms, you should open a bug report in the RN repo.

**Comment by j320** [comment]
Sorry, `tabBarAccessibilityLabel`, it's used in order to set an accessibility label in a navigation bar section, the same way the tabBarTestID is used to set a testID. When this is set, the Debug View Hierarchy returns the view as "Is Accessibility Element" and the label is the one set in `tabBarAccessibilityLabel`.

**Comment by LeoNatan** [comment]
Please post the full error you get by Detox, including the hierarchy dump. Thanks

**Comment by LeoNatan** [comment]
Are you using `react-navigation`? If so, it’s most likely not even a real tab bar (`UITabBar`). This is most likely not an issue with Detox.

**Comment by j320** [comment]
Yes I am.

It's strange though because I can tap on it by label with the set tabBarAccessbilityLabel but I cannot expect its label when matching it by ID or a different locator on iOS.

**Comment by LeoNatan** [comment]
> Please post the full error you get by Detox, including the hierarchy dump. Thanks

**Comment by j320** [comment]
https://gist.github.com/j320/996a4306be7ef7b47ddffe04c4c9d7cd

**Comment by LeoNatan** [comment]
Hierarchy looks OK. I will take a look at this tomorrow. Perhaps something in the expectation is not set up correctly.

**Comment by j320** [comment]
Thanks. Just as an update, I'm seeing the exact same behaviour with toHaveId() and tabBarTestID
